### DISCUSSION TITLE
VoiceOver issues with iDevice app

### DISCUSSION THREAD

**Discussion by a-singer** [jamulussoftware/jamulus#3313] [Bug Investigation]
Hello,

I'm having a bit of difficulty using the Jamulus app with VoiceOver on iOS. There are quite a few issues, but I suspect there are actually very few underlying difficulties, or only one. Basically, in Jamulus, VoiceOver does not seem to be able to select anything, whether with touch or through the keyboard. To reproduce:

1.	Launch VoiceOver with either triple click home or from the accessibility menu, it seems to make no difference how it starts and I have rebooted the phone.

2.	Open Jamulus. 

3.	Double tap on connect/disconnect. This will bring up the connect/disconnect dialog, as expected.

4.	Tap anywhere on the screen. What should happen is for the screen reader to tell you what your finger just tapped and for that item, whatever it is, to take VoiceOver focus. It doesn't. What actually happens is that VO just says "connect", which I think is the top line of the window. 

5.	Another method of navigation is by swiping. Swipe to your right and you will hear items supposedly take focus in the dialog. I'm not sure if they do because, when I double tap, just for example, to invoke a text box, I hear "connect". 

Something odd is happening where the VO focus is constantly returning to the top of the screen.

Again, I'm not sure whether this is the same issue but some controls are also unusable. To find just one,

1.	Open Jamulus.

2.	Double tap on settings. 

3.	Be sure to be in the audio setup.

4.	double tap the auto box for the buffers so that this is unchecked. This works.

5.	Swipe to either buffer. To change a value like this one, you usually just swipe up to increase, swipe down to decrease. In this case, I can do that as much as I like, but it still says "ten". I'm not sure whether it is actually changing and not being announced, or not changing at all. I suspect the latter since when I close the app and open it again, the buffer remains at ten.

I realize this isn't used very often, but any help would be very gratefully received.

**Comment by ann0see** [comment]
Edited your message to remove code formatting.

**Comment by a-singer** [comment]
Just to answer the question in 3311, this doesn't happen in the OSX Jamulus. There are quite a few accessibility issues in that one but they don't make the program unusable or prevent navigation as they do on iOS. I haven't tried the Windows or Linux version.

**Comment by ann0see** [comment]
If it isn't a problem on macOS, I'd assume that it's a QT (version) related issue. iOS is using Qt 5 at the moment since there are GUI issues on the Qt6 build. Upgrading requires a considerable amount of work but must happen eventually

**Comment by a-singer** [comment]
Thanks, this is most useful. Again, I understand that nobody is really interested in this particular use case, so it'll be upgraded whenever it's upgraded. If I may ask, though, is there a way to launch the app with a specific command line option to connect to a server? It's almost unusable as it stands, but once connected, it's fine. What I mean is something like the -c argument on the OSX or Windows command line to connect to a server. I understand this can be done with a URL, but am not sure whether that's available in the .IPA.

**Reply by ann0see** [reply]
It's not possible without compiling the ipa yourself, I'm afraid. Then you'd add the server in question to the code and set the -c flag.

**Comment by ann0see** [comment]
Converted to issue https://github.com/jamulussoftware/jamulus/issues/3315 since it's more of a bug.

**Reply by ann0see** [reply]
Please continue discussion there.
